include/dem/particle_wall_linear_force.h

```diff
@@ -193,19 +239,126 @@ class ParticleWallLinearForce
     return rolling_resistance_torque;
   }
 
+  /**
+   * @brief alculation of the rolling resistance torque model using the
+   * physical properties and the elastic-plastic spring-dashpot model
+   *
+   * @param[in] particle_properties Particle one properties.
+   * @param[in] effective_rolling_friction_coefficient Effective rolling
+   * friction coefficient.
+   * @param[in] effective_rolling_viscous_damping_coefficient Effective rolling
+   * viscous damping coefficient.
+   * @param[in] normal_force_norm Normal force norm.
+   * @param[in] dt DEM time step.
+   * @param[in] normal_spring_constant Normal spring constant.
+   * @param[in] normal_unit_vector Normal unit contact vector between the
+   * particle and the wall.
+   * @param[in,out] cumulative_rolling_resistance_spring_torque Cumulative
+   * rolling resistance torque for the EPSD rolling resistance model.
+   *
+   * @return Elasto-platique spring-dashpot rolling resistance torque
+   */
+  inline Tensor<1, 3>
+  epsd_resistance(const ArrayView<const double> &particle_properties,
```

Comment:
Yeah clearly that's a lot of copy-pase that sucks... We'll have to fix that for sure.